viewer/src/utilities/RxCounter.test.ts

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { RxCounter } from './RxCounter';
+import { interval } from 'rxjs';
+import { take, flatMap } from 'rxjs/operators';
+
+describe('RxCounter', () => {
+  test('increment on next', done => {
+    const counter = new RxCounter();
+    const incrementCount = 4;
+    expect.assertions(incrementCount);
+    const increment$ = interval(10).pipe(take(incrementCount), counter.incrementOnNext());
+    let expectedCount: number = 0;
+    counter.observeCount().subscribe({
+      next: count => {
+        expectedCount++;
+        expect(count).toBe(expectedCount);
+      }
+    });
+    increment$.subscribe({
+      complete: () => {
+        done();
+      }
+    });
+  });
+
+  test('decrement on next', done => {
+    // const incrementCount = 3;
+    // const decrementCount = 2;
+    const counter = new RxCounter();
+    const operationCount = 4;
+    const wait = flatMap(value => new Promise(resolve => setTimeout(resolve, 100, value)));
+    expect.assertions(operationCount * 2);
```

Comment:
What is this assertion?

Reply:
Assertion is a way to make sure the test has x numbers of assertions. It is a way to make sure that your loops etc and the number of tests that you assert are correct. In this case since we expect the increment to happen 4 times and in the subscriber we expect 4 values, assertions basically counts the number of expects and if it for some reason is more than 4 it fails it.